### ISSUE TITLE
 java.lang.SecurityException  com.onesignal.JobIntentService$JobServiceEngineImpl.dequeueWork

### ISSUE THREAD

**Issue by AndroidZbb** [OneSignal/OneSignal-Android-SDK#673]
**Description:**

java.lang.SecurityException

com.onesignal.JobIntentService$JobServiceEngineImpl.dequeueWork

(your description here)

**Environment**

3.10.3

Android 8.0  Android 8.1

**Steps to Reproduce Issue:**

1. (your steps here)
2.
3.

**Anything else:**

(crash stacktraces, as well as any other information here)
```java
java.lang.RuntimeException: 
  at android.os.AsyncTask$3.done (AsyncTask.java:365)
  at java.util.concurrent.FutureTask.finishCompletion (FutureTask.java:383)
  at java.util.concurrent.FutureTask.setException (FutureTask.java:252)
  at java.util.concurrent.FutureTask.run (FutureTask.java:271)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1162)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:636)
  at java.lang.Thread.run (Thread.java:784)
Caused by: java.lang.SecurityException: 
  at android.os.Parcel.readException (Parcel.java:1946)
  at android.os.Parcel.readException (Parcel.java:1892)
  at android.app.job.IJobCallback$Stub$Proxy.dequeueWork (IJobCallback.java:191)
  at android.app.job.JobParameters.dequeueWork (JobParameters.java:196)
  at com.onesignal.JobIntentService$JobServiceEngineImpl.dequeueWork (JobIntentService.java:339)
  at com.onesignal.JobIntentService.d (JobIntentService.java:668)
  at com.onesignal.JobIntentService$CommandProcessor.doInBackground (JobIntentService.java:414)
  at com.onesignal.JobIntentService$CommandProcessor.doInBackground (JobIntentService.java:407)
  at android.os.AsyncTask$2.call (AsyncTask.java:345)
  at java.util.concurrent.FutureTask.run (FutureTask.java:266)
```

**Comment by kunalbhise** [comment]
Facing Similar crash on Android 8, Android 9 devices.
SDK version:
implementation 'com.onesignal:OneSignal:[3.9.1, 3.99.99]'

StackTraces:
JobIntentService.java line 339
com.onesignal.JobIntentService$JobServiceEngineImpl.dequeueWork

Fatal Exception: java.lang.RuntimeException: An error occurred while executing doInBackground()
       at android.os.AsyncTask$3.done(AsyncTask.java:353)
       at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
       at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
       at java.util.concurrent.FutureTask.run(FutureTask.java:271)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at java.lang.Thread.run(Thread.java:764)
Caused by java.lang.SecurityException: Caller no longer running, last stopped +115ms because: last work dequeued
       at android.os.Parcel.readException(Parcel.java:2005)
       at android.os.Parcel.readException(Parcel.java:1951)
       at android.app.job.IJobCallback$Stub$Proxy.dequeueWork(IJobCallback.java:191)
       at android.app.job.JobParameters.dequeueWork(JobParameters.java:208)
       at com.onesignal.JobIntentService$JobServiceEngineImpl.dequeueWork(JobIntentService.java:339)
       at com.onesignal.JobIntentService.dequeueWork(JobIntentService.java:668)
       at com.onesignal.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:414)
       at com.onesignal.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:407)
       at android.os.AsyncTask$2.call(AsyncTask.java:333)
       at java.util.concurrent.FutureTask.run(FutureTask.java:266)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at java.lang.Thread.run(Thread.java:764)

**Comment by jkasten2** [comment]
Thanks for the details. This looks to be the same error as this open issue on Google's tracker.
https://issuetracker.google.com/issues/63622293

@kunalbhise There is a difference as you are getting "last work dequeued" but the stack traces and other parts of the error do match up. Are you able to provide steps on reproducing the issue.

**Comment by kunalbhise** [comment]
@jkasten2 Have tested Notification delivery on my test devices but no visible crash observed. Additionally, My Crashlytics reports approx 70 crashes/day indicating 92% when App is in the background.

**Comment by CodeBoyChd** [comment]
I am facing the same issue. Getting 6300 crashes a day. It's a high priority issue for me.
Below are the logs:

```javascript
Caused by java.lang.SecurityException: Caller no longer running, last stopped +622ms because: last work dequeued
       at android.os.Parcel.readException(Parcel.java:1943)
       at android.os.Parcel.readException(Parcel.java:1889)
       at android.app.job.IJobCallback$Stub$Proxy.dequeueWork(IJobCallback.java:191)
       at android.app.job.JobParameters.dequeueWork(JobParameters.java:196)
       at com.onesignal.JobIntentService$JobServiceEngineImpl.dequeueWork(JobIntentService.java:339)
       at com.onesignal.JobIntentService.dequeueWork(JobIntentService.java:668)
       at com.onesignal.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:414)
       at com.onesignal.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:407)
       at android.os.AsyncTask$2.call(AsyncTask.java:333)
       at java.util.concurrent.FutureTask.run(FutureTask.java:266)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at java.lang.Thread.run(Thread.java:764)
```

**Comment by jkasten2** [comment]
@CodeBoyChd Thanks for reporting. Can you provide the following details?
1. Android SDK version in your app.
2. List of device models and counts affected.
2. List of Android version and counts affected.

**Comment by jkasten2** [comment]
@AndroidZbb @CodeBoyChd @kunalbhise We just released SDK version 3.10.4 with a fix for this. Feel free to reply if this does not help and we can reopen this issue.
